Use of NODE_ENV prone to clashing

As reported by Web3Uchiha on discord, they encountered the error `Cannot find module '/home/chirag/.vscode/extensions/node_modules/@ignored/solidity-language-server/out/index.js'`. It's because in our client we are using the `NODE_ENV` var to determine the server module path. We may consider changing the variable that we use, or use a different source of truth for determining we are in development mode.

--- a/.vscode/launch.json
+++ b/.vscode/launch.json
@@ -9,7 +9,9 @@
       "runtimeExecutable": "${execPath}",
       "args": ["--extensionDevelopmentPath=${workspaceRoot}/client"],
       "outFiles": ["${workspaceRoot}/client/out/**/*.js"],
-      "env": { "NODE_ENV": "development" },
+      "env": {
+        "VSCODE_NODE_ENV": "development"
+      },
       "preLaunchTask": {
         "type": "npm",
         "script": "watch"

--- a/client/src/setup/setupExtensionState.ts
+++ b/client/src/setup/setupExtensionState.ts
@@ -11,8 +11,8 @@ export function setupExtensionState(
   { sentryDsn }: { sentryDsn: string }
 ): ExtensionState {
   const environment =
-    process.env.NODE_ENV === "development"
-      ? process.env.NODE_ENV
+    process.env.VSCODE_NODE_ENV === "development"
+      ? process.env.VSCODE_NODE_ENV
       : "production";
 
   const serverModulePath = context.asAbsolutePath(

--- a/server/src/utils/index.ts
+++ b/server/src/utils/index.ts
@@ -57,5 +57,5 @@ export function normalizeSlashes(p: string) {
 }
 
 export function isTestMode() {
-  return process.env.NODE_ENV === "development";
+  return process.env.VSCODE_NODE_ENV === "development";
 }

--- a/test/protocol/package.json
+++ b/test/protocol/package.json
@@ -6,7 +6,7 @@
   "scripts": {
     "postinstall": "cd ./projects/hardhat && npm install",
     "start": "tsc && node out/index.js",
-    "test": "cross-env NODE_ENV=development mocha",
+    "test": "cross-env VSCODE_NODE_ENV=development mocha",
     "lint": "npm run prettier -- --check && npm run eslint",
     "lint:fix": "npm run prettier -- --write && npm run lint -- --fix",
     "eslint": "eslint --max-warnings 0 \"./src/**/*.ts\" \"./test/**/*.ts\"",

--- a/test/runTests.ts
+++ b/test/runTests.ts
@@ -31,7 +31,7 @@ import { runTests } from "@vscode/test-electron";
         "--force-disable-user-env",
       ],
       extensionTestsEnv: {
-        NODE_ENV: "development",
+        VSCODE_NODE_ENV: "development",
       },
     });
   } catch (err) {

